src/library.js

```diff
@@ -583,6 +583,32 @@ LibraryManager.library = {
 #endif // USE_PTHREADS
   },
 
+#if MINIMAL_RUNTIME
+
+// If USES_DYNAMIC_ALLOC is not defined, do not compile in sbrk() or brk(), so that user gets a linker error if they attempt
+// to call into malloc() that would sbrk().
+#if USES_DYNAMIC_ALLOC
+  // If building with minimal runtime in release mode, where malloc() failures are not expected to throw and memory growth
+  // is not allowed, use a really small stub for sbrk() and brk() that return failure.
```

Comment:
These functions don't seem to return failure (-1)?

Reply:
Added support for checking failure. In wasm if this allocates past the end of the heap, then wasm load/store will trap, so there will be an explicit failure afterwards. In applications that run with very deterministic memory allocations (e.g. allocate everything at startup), the checks are redundant, so can be removed.